### DISCUSSION TITLE
Targets, tar_map, Directories with dates

### DISCUSSION THREAD

**Discussion by decat7** [ropensci/tarchetypes#122] [Help] [answered]
First of all thanks Will for an awesome package. I am issues with directories with dates being converted into as.numeric dates only in the context of running the pipeline.

Reprex:

**R/make_dataset.R**
```
make_dataset <- function(path,data,date_fun){
  require("data.table")

  if(!dir.exists(file.path(path,date_fun))){
    dir.create(file.path(path,date_fun),recursive = TRUE)
  }

  fwrite(data,file.path(path,date_fun,"data.csv"))

  return(file.path(path,date_fun))
}
```

**Run outside of target:**
```
dat = data.frame(x = sample.int(100), y = sample.int(100),sample(letters[1:2],100,replace = TRUE))

make_dataset(path = "./data/preexist/data_source1",data = dat,date_fun = as.Date("2022-11-04"))
```
**Output:**
_"./data/preexist/data_source1/2022-11-04"_
 
```make_dataset(path = "./data/preexist/data_source1",data = dat,date_fun = "2022-11-04")```
**Output:**
_"./data/preexist/data_source1/2022-11-04"_
 
The output is the same for as.Date and no as.Date and is the desired effect.

**Run with of targets and tar_map with static branching:**
__targets.R_

```
library(targets)
library(tarchetypes)
library(tidyr)
library(data.table)

source("R/make_dataset.R")
tar_option_set(packages = c("dplyr","data.table"))

dat = data.frame(x = sample.int(100), y = sample.int(100),sample(letters[1:2],100,replace = TRUE))

values <- tibble(
  target_id = c("data_source1_2022_11_04","data_source1_2022_11_11"),
  make_dataset_function = rlang::syms(c("make_dataset", "make_dataset")),
  summary_function = rlang::syms(c("make_summary", "make_summary")),
  date_fun = c("2022-11-04","2022-11-11")%>%as.Date(),
  input_path = c("./data/preexist/data_source1", "./data/preexist/data_source1")
)

targets <- tar_map(
  values = values,
  names = "target_id",
  tar_target(input,input_path,format = "file"),
  tar_target(data, make_dataset_function(input, dat,date_fun))
  #,tar_target(summary, summary_function(data))
)
list(targets)
```

`tar_make()`
![image](https://user-images.githubusercontent.com/84047973/201494530-23978684-b34b-468e-96fb-30e1719dbd91.png)

The folders are created as as.numeric(as.Date("2022-11-04")) which is 19300 instead of "2022-11-04"
![image](https://user-images.githubusercontent.com/84047973/201494519-16b3cf90-d238-4700-b359-39018914779b.png)

**It is a requirement that my function arg date_fun be a date data type and not a character.** `date_fun = c("2022-11-04","2022-11-11")%>%as.Date(),` in the values tibble. Changing date_fun to character will cause problems with other date calculations in more complex function steps. date_fun is would be used for more than just directory creation.

It is unclear if this is a tibble dates issues or targets issues in write date as a folder names. Any thoughts on what area is forcing as.numeric(as.Date("2022-11-04"))?

Thanks in advance!

**Comment by wlandau** [comment, marked as answer]
This is an instance of https://github.com/ropensci/tarchetypes/issues/94#issuecomment-1133199797. You could try substituting expressions instead of the actual data values.

```r
values <- tibble(
  ...,
  date_fun = purrr::map(c("2022-11-04","2022-11-11"), ~rlang::call2("as.Date", .x))
)
```

> It is a requirement that my function arg date_fun be a date data type and not a character.

There are several workarounds in R that may allow you to relax this requirement: for example, a wrapper around `make_dataset()` that converts a supplied character string to a date before calling `make_dataset()`.

**Reply by decat7** [reply]
Thanks for explaining. This workaround is fine.
